tools/sift.go

```diff
@@ -0,0 +1,565 @@
+	listInvestigations,
+)
+
+// FindErrorPatternLogsParams defines the parameters for running an ErrorPatternLogs check
+type FindErrorPatternLogsParams struct {
+	Name     string            `json:"name" jsonschema:"required,description=The name of the investigation"`
+	Labels   map[string]string `json:"labels" jsonschema:"required,description=Labels to scope the analysis"`
+	Start    time.Time         `json:"start,omitempty" jsonschema:"description=Start time for the investigation. Defaults to 30 minutes ago if not specified."`
+	End      time.Time         `json:"end,omitempty" jsonschema:"description=End time for the investigation. Defaults to now if not specified."`
+	QueryURL string            `json:"queryUrl,omitempty" jsonschema:"description=Optional query URL for the investigation"`
+}
+
+// findErrorPatternLogs creates an investigation with ErrorPatternLogs check, waits for it to complete, and returns the analysis
+func findErrorPatternLogs(ctx context.Context, args FindErrorPatternLogsParams) (*Analysis, error) {
+	client, err := siftClientFromContext(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("creating Sift client: %w", err)
+	}
+
+	// Create the investigation request with ErrorPatternLogs check
+	requestData := InvestigationRequest{
+		Labels:   args.Labels,
+		Start:    args.Start,
+		End:      args.End,
+		QueryURL: args.QueryURL,
+		Checks:   []string{string(CheckTypeErrorPatternLogs)},
+	}
+
+	investigation := &Investigation{
+		Name:       args.Name,
+		GrafanaURL: client.url,
+		Status:     InvestigationStatusPending,
+	}
+
+	// Create the investigation and wait for it to complete
+	completedInvestigation, err := client.createInvestigation(ctx, investigation, requestData)
+	if err != nil {
+		return nil, fmt.Errorf("creating investigation: %w", err)
+	}
+
+	// Get all analyses from the completed investigation
+	analyses, err := client.getAnalyses(ctx, completedInvestigation.ID)
+	if err != nil {
+		return nil, fmt.Errorf("getting analyses: %w", err)
+	}
+
+	// Find the ErrorPatternLogs analysis
+	var errorPatternLogsAnalysis *Analysis
+	for i := range analyses {
+		if analyses[i].Name == string(CheckTypeErrorPatternLogs) {
+			errorPatternLogsAnalysis = &analyses[i]
+			break
+		}
+	}
+
+	if errorPatternLogsAnalysis == nil {
+		return nil, fmt.Errorf("ErrorPatternLogs analysis not found in investigation %s", completedInvestigation.ID)
+	}
+
+	return errorPatternLogsAnalysis, nil
+}
+
+// FindErrorPatternLogs is a tool for running an ErrorPatternLogs check
+var FindErrorPatternLogs = mcpgrafana.MustTool(
+	"find_error_pattern_logs",
+	"Creates an investigation to search for error patterns in logs, waits for it to complete, and returns the analysis results. This tool triggers an investigation in the relevant Loki datasource to determine if there are elevated error rates compared to the last day's average, and returns the error pattern found, if any.",
```

Comment:
Shall we get rid of mentions of an 'investigation' here too, so it looks like the tool just opaquely finds error patterns in logs? (the implementation isn't super important imo)

Same with slow requests